src/index.ts

```diff
@@ -860,6 +908,22 @@ class Piscina extends EventEmitterAsyncResource {
   static get Piscina () {
     return Piscina;
   }
+
+  static move (val : any) {
+    if (!isTransferable(val)) {
+      if (types.isAnyArrayBuffer(val)) {
+        val = new DirectlyTransferable(val);
+      } else if ((types as any).isArrayBufferView(val)) {
+        val = new ArrayBufferViewTransferable(val as ArrayBufferView);
+      } else if (val instanceof MessagePort) {
+        val = new DirectlyTransferable(val);
+      } else {
+        throw new TypeError('value is not transferable');
+      }
+    }
```

Comment:
Fwiw, `instanceof MessagePort` is an incomplete brand check (because it’s perfectly fine for ports to live in different Contexts), and generally there will not be an exhaustive list of types to check against – for example, we are adding support for [`KeyObject`](https://github.com/nodejs/node/pull/33360) right now.

That’s tricky and I’m not really sure how to work around that.

Reply:
I would probably just allow any kind of object here, and if it’s not actually transferable, then so be it.

We can still improve the experience for TypeScript users here if we define this as 

```js
static move (val : Transferable | TransferListItem | ArrayBufferView) {
  ...
```

where 

```js
type TransferListItem = TransferList extends (infer T)[] ? T : never;
```

Reply:
Yeah, I'm not happy with this one either. What would be interesting is a `isTransferable()` in *core* that matches whatever set of transferables that version of Node.js supports (e.g. I'm not certain the `KeyObject` one will be backported any time soon).

Btw, is the `KeyObject` one actually going to allow the key to be transferred or just cloned efficiently?

Reply:
Right, core is more or less the only authoritative source on whether something can be transferred or not.

The `KeyObject` one supports cloning in its current state, but it could also easily be extended to transferring.

Reply:
And yeah, relaxing it to any object is likely best for now.